### PR TITLE
[website] restore javadoc generation and automate website deployment

### PR DESCRIPTION
**Pull request by nicoloboschi**
### Motivation
With gradle the Javadoc generation is not configured well (or at least as it was in maven).
The website is not automatically built with the latest changes. According to https://github.com/apache/bookkeeper/pull/3080 the website deployment will be handled by the ASF infra.
  

### Changes
* Restored Javadoc configuration from the [maven one](https://github.com/apache/bookkeeper/blob/release-4.13.0/pom.xml#L897) 
* Added the javadoc generation to the CI to ensure it will not break
* Added a periodic job (once per day) to build and publish the website (if any changes occur)
Please note that the first time the job will run, it will update the website that has been unmodified for almost a year.

### CODE REVIEW

**Comment by nicoloboschi** [comment]
rerun failure checks

File: site/scripts/javadoc-gen.sh [hunk 1]
```diff
@@ -7,18 +7,31 @@ function build_javadoc() {
...
+    fi
+
   fi
   javadoc_dest_dir="${ROOT_DIR}/site/docs/${version}/api/javadoc"
   rm -rf $javadoc_dest_dir
-  mvn clean install javadoc:aggregate -DskipTests
+  if [[ "$use_gradle" == "true" ]]; then
+    ./gradlew generateApiJavadoc
+  else
+    mvn clean install javadoc:aggregate -DskipTests
```
Comment (dlg99) [inline site/scripts/javadoc-gen.sh:32]: maven is gone, we can get rid of this and of the code related to "use_gradle"

**Review by dlg99 — APPROVED** [review]
LGTM but let's remove unnecessary maven stuff

File: site/scripts/javadoc-gen.sh (hunk 1 above)
Comment (nicoloboschi) [reply site/scripts/javadoc-gen.sh:32]: this script it is intended to also build older bk versions

**Review by eolivelli — APPROVED** [review]

**Pull request merged**